File: samples/cpp/authenticate.cc
// License: Apache 2.0. See LICENSE file in root directory.
// Copyright(c) 2020-2021 Intel Corporation. All Rights Reserved.

#include "RealSenseID/FaceAuthenticator.h"
#include <iostream>

class MyAuthClbk : public RealSenseID::AuthenticationCallback
{
public:
    void OnResult(const RealSenseID::AuthenticateStatus status, const char* user_id) override
    {
        if (status == RealSenseID::AuthenticateStatus::Success)
            std::cout << "Authenticated " << user_id << std::endl;
    }

    void OnHint(const RealSenseID::AuthenticateStatus hint) override
    {
        std::cout << "OnHint " << hint << std::endl;
    }

    void OnFaceDetected(const std::vector<RealSenseID::FaceRect>& faces) override
    {
        for (auto& face : faces)
        {
            std::cout << "Detected face " << face.x << "," << face.y << " " << face.w << "x" << face.h << std::endl;
        }
    }
};

int main()
{
    RealSenseID::FaceAuthenticator authenticator;
    auto status = authenticator.Connect({"COM9"});
    if (status != RealSenseID::Status::Ok)
    {
        std::cout << "Failed connecting with status " << status << std::endl;
        return 1;
    }
    MyAuthClbk auth_clbk;
    authenticator.Authenticate(auth_clbk);
}

File: samples/python/authenticate.py
"""
License: Apache 2.0. See LICENSE file in root directory.
Copyright(c) 2020-2021 Intel Corporation. All Rights Reserved.
"""

import rsid_py

PORT='COM9'

def on_result(result, user_id):
    print('on_result', result)    
    if result == rsid_py.AuthenticateStatus.Success:
        print('Authenticated user:', user_id)

def on_faces(faces, timestamp):    
    print(f'detected {len(faces)} face(s)')
    for f in faces:
        print(f'\tface {f.x},{f.y} {f.w}x{f.h}')    

if __name__ == '__main__':
    with rsid_py.FaceAuthenticator(PORT) as f:
        f.authenticate(on_faces=on_faces, on_result=on_result)
    